Description: include net/bpf.h on kfreebsd to enable bpf
Author: Romain Francoise <rfrancoise@debian.org>
Origin: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=619684
Index: ns2/emulate/net-pcap.cc
===================================================================
--- ns2.orig/emulate/net-pcap.cc	2011-08-10 14:57:34.480781056 +0800
+++ ns2/emulate/net-pcap.cc	2011-08-10 15:02:17.378183867 +0800
@@ -55,6 +55,10 @@
 #include <sys/systeminfo.h>
 #endif
 
+#if !defined(__linux__)&&!defined(__APPLE__)
+#include <net/bpf.h>
+#endif
+
 #ifdef __cplusplus
 extern "C" {
 #include <pcap.h>
