Index: ns2/mac/mac-802_11Ext.h
===================================================================
--- ns2.orig/mac/mac-802_11Ext.h	2011-08-10 15:45:35.963069556 +0800
+++ ns2/mac/mac-802_11Ext.h	2011-08-10 15:46:40.403389096 +0800
@@ -57,6 +57,7 @@
  
 #ifndef ns_mac_80211Ext_h
 #define ns_mac_80211Ext_h
+#include <cstddef>
 #include "marshall.h"
 #include "timer-handler.h"
 #define GET_ETHER_TYPE(x)		GET2BYTE((x))
