Index: ns2/nix/nixnode.h
===================================================================
--- ns2.orig/nix/nixnode.h	2012-06-03 17:49:42.528663107 +0800
+++ ns2/nix/nixnode.h	2012-06-03 17:54:09.000000000 +0800
@@ -9,7 +9,7 @@
 #include "routealgo/rnode.h"
 #include "object.h"
 #include <map>
-
+using namespace std;
 // Define the edge class
 class Edge {
 public :
Index: ns2/nix/nixvec.h
===================================================================
--- ns2.orig/nix/nixvec.h	2012-06-03 17:49:42.528663107 +0800
+++ ns2/nix/nixvec.h	2012-06-03 17:54:09.000000000 +0800
@@ -57,7 +57,7 @@
 #ifdef WIN32
 #include <pair.h>   // for MSVC 6.0 that doens't have a proper <utility>
 #endif /* WIN32 */
-
+using namespace std;
 // Define a type for the neighbor index
 typedef unsigned long Nix_t;
 typedef unsigned long Nixl_t; // Length of a NV
Index: ns2/routealgo/rnode.h
===================================================================
--- ns2.orig/routealgo/rnode.h	2012-06-03 17:49:42.528663107 +0800
+++ ns2/routealgo/rnode.h	2012-06-03 17:54:09.000000000 +0800
@@ -15,7 +15,7 @@
 #ifdef USE_HASH
 #include <hash_set>
 #endif
-
+using namespace std;
 typedef unsigned long nodeid_t; // Node identifier
 typedef unsigned long dist_t;   // Distance
 typedef unsigned int  weight_t; // Weight
Index: ns2/mac/wireless-phyExt.h
===================================================================
--- ns2.orig/mac/wireless-phyExt.h	2012-06-03 17:49:42.528663107 +0800
+++ ns2/mac/wireless-phyExt.h	2012-06-03 17:54:09.000000000 +0800
@@ -66,7 +66,7 @@
 #include "timer-handler.h"
 #include <list>
 #include <packet.h>
-
+using namespace std;
 enum PhyState {SEARCHING = 0, PreRXing = 1, RXing = 2, TXing = 3};
 
 typedef struct ModulationParam {
Index: ns2/mobile/shadowing-vis.cc
===================================================================
--- ns2.orig/mobile/shadowing-vis.cc	2012-06-03 17:49:42.528663107 +0800
+++ ns2/mobile/shadowing-vis.cc	2012-06-03 17:54:09.000000000 +0800
@@ -71,7 +71,7 @@
 //#define WHITE 0xff
 
 //#define DEBUG
-
+using namespace std;
 static class ShadowingVisClass: public TclClass {
 public:
 	ShadowingVisClass() : TclClass("Propagation/ShadowingVis") {}
Index: ns2/mdart/mdart_function.h
===================================================================
--- ns2.orig/mdart/mdart_function.h	2012-06-03 17:49:42.528663107 +0800
+++ ns2/mdart/mdart_function.h	2012-06-03 17:54:09.000000000 +0800
@@ -68,7 +68,7 @@
 
 
 #include <mdart/mdart_packet.h>
-
+using namespace std;
 
 
 //------------------------------------------------------------------------------
Index: ns2/mpls/ldp.cc
===================================================================
--- ns2.orig/mpls/ldp.cc	2012-06-03 17:49:42.528663107 +0800
+++ ns2/mpls/ldp.cc	2012-06-03 17:54:09.000000000 +0800
@@ -69,7 +69,7 @@
 #include <iostream>
 
 #include "ldp.h"
-
+using namespace std;
 int hdr_ldp::offset_;
 
 static class LDPHeaderClass : public PacketHeaderClass {
Index: ns2/asim/asim.cc
===================================================================
--- ns2.orig/asim/asim.cc	2012-06-03 17:49:42.528663107 +0800
+++ ns2/asim/asim.cc	2012-06-03 17:54:09.000000000 +0800
@@ -61,7 +61,7 @@
 #include "asim.h"
 
 #define sp " " 
-
+using namespace std;
 typedef struct c{
   int no; // no of edges in the connection
   double delay; // total delay;
Index: ns2/routealgo/rbitmap.h
===================================================================
--- ns2.orig/routealgo/rbitmap.h	2012-06-03 17:49:42.528663107 +0800
+++ ns2/routealgo/rbitmap.h	2012-06-03 17:54:09.000000000 +0800
@@ -59,7 +59,7 @@
 
 #include <iostream>
 #include <sys/types.h>
-
+using namespace std;
 const unsigned  int BITS_ULONG = 32;
 
 class BitMap {
Index: ns2/common/scheduler-map.cc
===================================================================
--- ns2.orig/common/scheduler-map.cc	2012-06-03 17:49:42.528663107 +0800
+++ ns2/common/scheduler-map.cc	2012-06-03 17:54:09.000000000 +0800
@@ -43,7 +43,7 @@
 #include <set>
 
 #include "scheduler.h"
-
+using namespace std;
 
 class MapScheduler : public Scheduler {
 public:
Index: ns2/linkstate/ls.cc
===================================================================
--- ns2.orig/linkstate/ls.cc	2012-06-03 17:49:42.528663107 +0800
+++ ns2/linkstate/ls.cc	2012-06-03 17:54:09.000000000 +0800
@@ -65,7 +65,7 @@
 #ifdef HAVE_STL
 
 #include "ls.h"
-
+using namespace std;
 // a global variable
 LsMessageCenter LsMessageCenter::msgctr_;
 
Index: ns2/linkstate/ls.h
===================================================================
--- ns2.orig/linkstate/ls.h	2012-06-03 17:49:42.528663107 +0800
+++ ns2/linkstate/ls.h	2012-06-03 17:54:09.000000000 +0800
@@ -70,7 +70,7 @@
 #include <utility>
 
 #include "timer-handler.h"
-
+using namespace std;
 const int LS_INVALID_COUNT = -1;
 const int LS_INIT_ACCESS_COUNT = 3;
 const int LS_INVALID_NODE_ID = 65535;
@@ -134,7 +134,7 @@
 		return ib.second ? ib.first : baseMap::end();
 	}
 
-	void eraseAll() { erase(baseMap::begin(), baseMap::end()); }
+	void eraseAll() { baseMap::erase(baseMap::begin(), baseMap::end()); }
 	T* findPtr(Key key) {
 		iterator it = baseMap::find(key);
 		return (it == baseMap::end()) ? (T *)NULL : &((*it).second);
Index: ns2/pgm/pgm-agent.cc
===================================================================
--- ns2.orig/pgm/pgm-agent.cc	2012-06-03 17:49:42.528663107 +0800
+++ ns2/pgm/pgm-agent.cc	2012-06-03 17:54:09.000000000 +0800
@@ -74,7 +74,7 @@
 #include "basetrace.h"
 
 #include "pgm.h"
-
+using namespace std;
 // ************************************************************
 // Register the PGM packet headers.
 // ************************************************************
Index: ns2/pgm/pgm-sender.cc
===================================================================
--- ns2.orig/pgm/pgm-sender.cc	2012-06-03 17:49:42.528663107 +0800
+++ ns2/pgm/pgm-sender.cc	2012-06-03 17:54:09.000000000 +0800
@@ -75,7 +75,7 @@
 #include "basetrace.h"
 
 #include "pgm.h"
-
+using namespace std;
 // ************************************************************
 // Define the PGM Sender Timer Class
 // ************************************************************
Index: ns2/pgm/pgm-receiver.cc
===================================================================
--- ns2.orig/pgm/pgm-receiver.cc	2012-06-03 17:49:42.528663107 +0800
+++ ns2/pgm/pgm-receiver.cc	2012-06-03 17:54:09.000000000 +0800
@@ -74,7 +74,7 @@
 #include "rcvbuf.h"
 
 #include "pgm.h"
-
+using namespace std;
 // ************************************************************
 // Define the PGM Receive Timer Class
 // ************************************************************
Index: ns2/diffusion3/ns/difftimer.h
===================================================================
--- ns2.orig/diffusion3/ns/difftimer.h	2011-04-15 17:26:28.000000000 +0800
+++ ns2/diffusion3/ns/difftimer.h	2012-06-03 18:02:00.040320228 +0800
@@ -56,7 +56,7 @@
 #include "scheduler.h"
 #include "timers.hh"
 
-
+using namespace std;
 class MapEntry;
 class DiffEventQueue;
 
Index: ns2/delaybox/delaybox.h
===================================================================
--- ns2.orig/delaybox/delaybox.h	2011-04-15 17:26:28.000000000 +0800
+++ ns2/delaybox/delaybox.h	2012-06-03 18:39:41.687535054 +0800
@@ -43,7 +43,7 @@
 #include "node.h"
 #include "ranvar.h"
 #include <map>
-
+using namespace std;
 class DelayBoxClassifier;
 
 void packet_string (char* str, hdr_tcp* tcph, hdr_ip* iph, int size);
Index: ns2/packmime/packmime_HTTP.h
===================================================================
--- ns2.orig/packmime/packmime_HTTP.h	2011-04-15 17:26:28.000000000 +0800
+++ ns2/packmime/packmime_HTTP.h	2012-06-03 18:41:01.471930762 +0800
@@ -54,7 +54,7 @@
 #include <map>
 
 #define MAX_NODES 10 
-
+using namespace std;
 class FullTcpAgent;
 class PackMimeHTTP;
 class PackMimeHTTPTimer;
Index: ns2/packmime/packmime_OL.h
===================================================================
--- ns2.orig/packmime/packmime_OL.h	2011-04-15 17:26:28.000000000 +0800
+++ ns2/packmime/packmime_OL.h	2012-06-03 19:15:41.958247325 +0800
@@ -73,7 +73,7 @@
 	void init();
 	int size_;
 	int seqno_;
-	int64_t maxpkts_;	
+	int maxpkts_;
 	
 	// statistics objects
 	RandomVariable* pm_pac_ia_;
Index: ns2/tmix/tmix.h
===================================================================
--- ns2.orig/tmix/tmix.h	2011-10-24 16:05:24.507039988 +0800
+++ ns2/tmix/tmix.h	2012-06-03 20:32:06.623570085 +0800
@@ -68,7 +68,7 @@
 #define CVEC_LINE_MAX 100
 #define CV_V1 1
 #define CV_V2 2
-
+using namespace std;
 class FullTcpAgent;
 class Tmix;
 class TmixTimer;
